doc/source/dev/index.rst

```diff
@@ -60,6 +60,14 @@ Here's the short summary, complete TOC links are below:
      - ``upstream``, which refers to the ``numpy`` repository
      - ``origin``, which refers to your personal fork
 
+   * Get the tag information from the upstream repository::
+
+      git pull upstream --tags
+
+   * Initialize numpy's submodules::
+
+      git submodule update --init
```

Comment:
This should go after pulling from upstream main below. Might move pulling the tags also.